### PR TITLE
add grpc-js support, add filter, add tests

### PR DESCRIPTION
**Pull request by prodion23**
## Description
This adds `@grpc/grpc-js` support.  

### Note about `grpc` vs `@grpc/grpc-js`
There are 2 grpc packages, `@grpc/grpc-js` is the newer, non-deprecated one. 
The other one is here: https://www.npmjs.com/package/grpc
I could not get auto instrumentation to work for the client side when using the deprecated package. 

### .original files
There are no hooks exposed for otel grpc support, unfortunately since a lot of the places we need hooks are within unexported anonymous functions, we can't just replace them, so instead the `.original` is the original otel instrumentation source, the file with `Hypertrace` in it has our modifications.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -15,8 +15,8 @@
   "dependencies": {
     "@opentelemetry/api": "1.0.4",
     "@opentelemetry/core": "1.0.1",
-    "@opentelemetry/exporter-trace-otlp-http": "^0.27.0",
```
Comment (tim-mwangi) [inline package.json:18]: lots of changes in the package-lock.json above

File: src/HypertraceAgent.ts [hunk 2]
```diff
@@ -24,13 +25,12 @@ import {GraphQLInstrumentation} from "@opentelemetry/instrumentation-graphql";
 import {logger} from "./Logging";
 import {version} from "./Version";
 import {OTLPTraceExporter} from "@opentelemetry/exporter-trace-otlp-grpc";
-import {HttpInstrumentation} from "@opentelemetry/instrumentation-http";
```
Comment (tim-mwangi) [inline src/HypertraceAgent.ts:27]: http was removed?

Comment (prodion23) [reply src/HypertraceAgent.ts:27]: we don't use default otel http instrumentation(changes for filters) - this was just an unused import.

Comment (tim-mwangi) [reply src/HypertraceAgent.ts:27]: ok

**Review by tim-mwangi — APPROVED** [review]

**Pull request merged**